bootstrap/terraform/main.tf

```diff
@@ -1,198 +1,147 @@
-  enable_dns_hostnames = true
-  single_nat_gateway   = true
-
-  public_subnet_tags = {
-    "kubernetes.io/role/elb" = "1"
-  }
-
-  private_subnet_tags = {
-    "kubernetes.io/role/internal-elb" = "1"
+  cluster_name    = local.name
+  
+  crossplane_helm_config = var.crossplane_helm_config
+  crossplane_aws_provider = var.crossplane_aws_provider
+  crossplane_kubernetes_provider = var.crossplane_kubernetes_provider
+  
+  vpc_name        = local.name
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 2)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/awslabs/crossplane-on-eks"
   }
 }
 
 #---------------------------------------------------------------
-# This module deploys EKS Cluster with one Managed group
+# EKS Blueprints
 #---------------------------------------------------------------
+
 module "eks_blueprints" {
   source = "github.com/aws-ia/terraform-aws-eks-blueprints"
 
+  # EKS CONTROL PLANE VARIABLES
+  cluster_name    = local.cluster_name
+  cluster_version = local.cluster_version
+
   # EKS Cluster VPC and Subnet mandatory config
   vpc_id             = module.vpc.vpc_id
   private_subnet_ids = module.vpc.private_subnets
 
-  # EKS CONTROL PLANE VARIABLES
-  create_eks      = true
-  cluster_version = local.cluster_version
-  cluster_name    = local.cluster_name
-
   # EKS MANAGED NODE GROUPS
   managed_node_groups = {
-    mg_5 = {
-      node_group_name = local.node_group_name
-      instance_types  = ["m5.xlarge"]
-      min_size        = "1"
+    mg = {
+      node_group_name = "managed-spot"
+      capacity_type   = "SPOT"
+      instance_types  = ["t3.small", "t3a.small", "t3.medium"]
+      min_size        = 1
+      max_size        = 3
+      desired_size    = 2
       subnet_ids      = module.vpc.private_subnets
-      additional_tags = {
-        ExtraTag    = "m5-on-demand"
-        Name        = "m5-on-demand"
-        subnet_type = "private"
-      }
     }
   }
+
+  tags = local.tags
 }
 
 #---------------------------------------------------------------
-# This module deploys Kubernetes add-ons
+# EKS Blueprints Addons
 #---------------------------------------------------------------
-module "eks_blueprints_kubernetes_addons" {
+
+ module "eks_blueprints_kubernetes_addons" {
   source         = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons"
-  eks_cluster_id = module.eks_blueprints.eks_cluster_id
 
-  # Deploy Karpenter Autoscaler
-  enable_karpenter = true
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
 
   # Deploy Crossplane
   enable_crossplane = true
 
-  crossplane_helm_config = {
-    name       = "crossplane"
-    chart      = "crossplane"
-    repository = "https://charts.crossplane.io/stable/"
-    version    = "1.10.1"
-    namespace  = "crossplane-system"
-    values = [templatefile("${path.module}/values.yaml", {
-      operating-system = "linux"
-    })]
-  }
```

Comment:
I moved it to the variables.tf

Reply:
they can see in `variables.tf` and example if they want to edit `values.yaml` and how to pass it in

Reply:
I like your approach of moving to a variable for helm config however defining the 

`values = [templatefile("${path.module}/values.yaml", {
      operating-system = "linux"
    })]`

 is not possible through variable default values. Kubernetes users likes to use yaml for the configuration instead of formatted templates using terraform.

Reply:
Since we are consuming the EKS Blueprints module so we can just leave the `crossplane_helm_config` inside the code without introducing a new variable.